responses/bokbok.go

```diff
@@ -0,0 +1,212 @@
+package responses
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"github.com/boltdb/bolt"
+	"github.com/perthgophers/puddle/messagerouter"
+	"io"
+	"log"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+// BokBok is a smart chat bot
+type BokBok struct {
+	db        *bolt.DB
+	prefixLen int
+}
+
+// NewBokBok makes a BoKBok and opens the bolt database
+func NewBokBok(prefixLen int) *BokBok {
+	var err error = nil
+	bkbk := new(BokBok)
+	bkbk.db, err = bolt.Open("./markovchains.db", 0600, &bolt.Options{Timeout: 1 * time.Second})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	bkbk.db.Update(func(tx *bolt.Tx) error {
+		tx.CreateBucketIfNotExists([]byte("all"))
+		return nil
+	})
+
+	bkbk.prefixLen = prefixLen
+
+	return bkbk
+}
+
+// Processes Message builds or amends the Markov Chain for "all" and the individual user
+func (bkbk *BokBok) ProcessMessage(cr *messagerouter.CommandRequest, w messagerouter.ResponseWriter) error {
+	bkbk.db.Update(func(tx *bolt.Tx) error {
+		tx.CreateBucketIfNotExists([]byte(cr.Username))
+		return nil
+	})
+	ch := bkbk.Chain(cr.Username)
+	ch.BuildFromString(cr.Text)
+
+	allch := bkbk.Chain("all")
+	allch.BuildFromString(cr.Text)
+
+	bkbk.SaveChains(ch, allch)
+	return nil
+}
+
+// MaybeRespond might respond, or it might not, for top kek
+func (bkbk *BokBok) MaybeRespond(cr *messagerouter.CommandRequest, w messagerouter.ResponseWriter) error {
+	if cr.Message.Channel == "C33C4MJSH" && bkbk.YesNo() {
```

Comment:
Actually, spamchannel is unexported.

Should mr.channel & mr.spamchannel be changed to mr.Channel & mr.SpamChannel respectively?